DQM/Physics/src/TopSingleLeptonDQM.cc

```diff
@@ -520,124 +553,75 @@ void MonitorEnsemble::fill(const edm::Event& event,
-             "cfg file:              \n"
-          << "                                                                 "
-             "                     \n"
-          << "  ## load jet corrections                                        "
-             "                     \n"
-          << "  "
-             "process.load(\"JetMETCorrections.Configuration."
-             "JetCorrectionServicesAllAlgos_cff\") \n"
-          << "  process.prefer(\"ak5CaloL2L3\")                                "
-             "                     \n"
-          << "                                                                 "
-             "                     \n"
-          << "-----------------------------------------------------------------"
-             "-------------------- \n";
-    }
-  }
+
+	edm::Handle<reco::JetCorrector> corrector;
+	event.getByToken(mJetCorrector, corrector);
 
   // loop jet collection
   std::vector<reco::Jet> correctedJets;
   std::vector<double> JetTagValues;
-  unsigned int mult = 0, multCSV = 0;
+  unsigned int mult = 0, multLoose = 0, multCSV = 0;
 
   edm::Handle<edm::View<reco::Jet> > jets;
   if (!event.getByToken(jets_, jets)) {
     return;
   }
 
-  edm::Handle<reco::JetIDValueMap> jetID;
-  if (jetIDSelect_) {
-    if (!event.getByToken(jetIDLabel_, jetID)) return;
-  }
-
   for (edm::View<reco::Jet>::const_iterator jet = jets->begin();
        jet != jets->end(); ++jet) {
-    // check jetID for calo jets
-    unsigned int idx = jet - jets->begin();
-    if (jetIDSelect_ &&
-        dynamic_cast<const reco::CaloJet*>(jets->refAt(idx).get())) {
-      if (!(*jetIDSelect_)((*jetID)[jets->refAt(idx)])) continue;
-    }
-    // chekc additional jet selection for calo, pf and bare reco jets
-    if (dynamic_cast<const reco::CaloJet*>(&*jet)) {
-      reco::CaloJet sel = dynamic_cast<const reco::CaloJet&>(*jet);
-      sel.scaleEnergy(corrector ? corrector->correction(*jet) : 1.);
-      StringCutObjectSelector<reco::CaloJet> jetSelect(jetSelect_);
-      if (!jetSelect(sel)) {
-        continue;
-      }
-    } else if (dynamic_cast<const reco::PFJet*>(&*jet)) {
+		bool isLoose = false;
+    // check additional jet selection for pf jets
+		if (dynamic_cast<const reco::PFJet*>(&*jet)) {
       reco::PFJet sel = dynamic_cast<const reco::PFJet&>(*jet);
-      sel.scaleEnergy(corrector ? corrector->correction(*jet) : 1.);
       StringCutObjectSelector<reco::PFJet> jetSelect(jetSelect_);
-      if (!jetSelect(sel)) continue;
-    } else {
-      reco::Jet sel = *jet;
-      sel.scaleEnergy(corrector ? corrector->correction(*jet) : 1.);
-      StringCutObjectSelector<reco::Jet> jetSelect(jetSelect_);
+      StringCutObjectSelector<reco::PFJet> jetlooseSelect("chargedHadronEnergyFraction()>0 && chargedMultiplicity()>0 && chargedEmEnergyFraction()<0.99 && neutralHadronEnergyFraction()<0.99 && neutralEmEnergyFraction()<0.99 && (chargedMultiplicity()+neutralMultiplicity())>1");
```

Comment:
both jetSelect and jetSelectLoose need to be redeclared as member data so that they are created once per job (per instance of the module)

Reply:
changed to 

for (edm::View<reco::Jet>::const_iterator jet = jets->begin();
       jet != jets->end(); ++jet) {
                bool isLoose = false;
    // check additional jet selection for pf jets
                if (dynamic_cast<const reco::PFJet*>(&*jet)) {
      reco::PFJet sel = dynamic_cast<const reco::PFJet&>(*jet);
                        if ((*jetlooseSelection_)(sel)) isLoose = true;
      sel.scaleEnergy(corrector->correction(*jet));
      if (!(*jetSelection_)(sel)) continue;
    }